File: src/main/java/fr/vlegall/sochief/service/status/StatusService.java
package fr.vlegall.sochief.service.status;

import fr.vlegall.sochief.contracts.response.ApiStatus;
import org.springframework.boot.info.BuildProperties;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Service;

@Service
public class StatusService implements IStatusService {
    private final Environment environment;
    private final BuildProperties build;

    public StatusService(Environment environment, BuildProperties build) {
        this.environment = environment;
        this.build = build;
    }

    @Override
    public ApiStatus getStatus() {
        String[] profiles = environment.getActiveProfiles();
        String profile = profiles.length > 0 ? profiles[0] : "Default";
        String version = build != null ? build.getVersion() : "dev";
        String name = build != null ? build.getName() : (environment.getProperty("spring.application.name") != null
                ? environment.getProperty("spring.application.name") : "unknown");
        return new ApiStatus(name, profile, version, true);
    }
}

File: src/main/kotlin/fr/vlegall/sochief/service/status/StatusService.kt
package fr.vlegall.sochief.service.status

import fr.vlegall.sochief.contracts.response.ApiStatus
import org.springframework.core.env.Environment
import org.springframework.boot.info.BuildProperties
import org.springframework.stereotype.Service

@Service
class StatusService(
    private val environment: Environment,
    private val build: BuildProperties?
): IStatusService {


    override fun getStatus(): ApiStatus {
        val profile = environment.activeProfiles.firstOrNull() ?: "Default"
        val version = build?.version ?: "dev"
        val name = build?.name ?: environment.getProperty("spring.application.name") ?: "unknown"

        return ApiStatus(name, profile, version, true)
    }
}